packages/flutter/test/material/flat_button_test.dart

```diff
@@ -334,79 +412,209 @@ void main() {
+              child: const Text('ABC'),
+            ),
+          ),
+        ),
+      ),
     );
-    flatButton = find.byType(FlatButton);
-    expect(tester.widget<FlatButton>(flatButton).enabled, true);
-    await tester.tap(flatButton);
-    expect(wasPressed, true);
 
-    // onPressed null, onLongPress not null.
-    wasPressed = false;
+    expect(semantics, hasSemantics(
+      TestSemantics.root(
+        children: <TestSemantics>[
+          TestSemantics.rootChild(
+            actions: <SemanticsAction>[
+              SemanticsAction.tap,
+            ],
+            label: 'ABC',
+            rect: const Rect.fromLTRB(0.0, 0.0, 88.0, 48.0),
+            transform: Matrix4.translationValues(356.0, 276.0, 0.0),
+            flags: <SemanticsFlag>[
+              SemanticsFlag.hasEnabledState,
+              SemanticsFlag.isButton,
+              SemanticsFlag.isEnabled,
+              SemanticsFlag.isFocusable,
+            ],
+          ),
+        ],
+      ),
+      ignoreId: true,
+    ));
+
+    semantics.dispose();
+  });
+
+  testWidgets('Does FlatButton scale with font scale changes', (WidgetTester tester) async {
     await tester.pumpWidget(
-      buildFrame(onPressed: null, onLongPress: () { wasPressed = true; }),
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Material(
+          child: MediaQuery(
+            data: const MediaQueryData(textScaleFactor: 1.0),
+            child: Center(
+              child: FlatButton(
+                onPressed: () { },
+                child: const Text('ABC'),
+              ),
+            ),
+          ),
+        ),
+      ),
     );
-    flatButton = find.byType(FlatButton);
-    expect(tester.widget<FlatButton>(flatButton).enabled, true);
-    await tester.longPress(flatButton);
-    expect(wasPressed, true);
 
-    // onPressed null, onLongPress null.
+    expect(tester.getSize(find.byType(FlatButton)), equals(const Size(88.0, 48.0)));
+    expect(tester.getSize(find.byType(Text)), equals(const Size(42.0, 14.0)));
+
+    // textScaleFactor expands text, but not button.
     await tester.pumpWidget(
-      buildFrame(onPressed: null, onLongPress: null),
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Material(
+          child: MediaQuery(
+            data: const MediaQueryData(textScaleFactor: 1.3),
+            child: Center(
+              child: FlatButton(
+                onPressed: () { },
+                child: const Text('ABC'),
+              ),
+            ),
+          ),
+        ),
+      ),
     );
-    flatButton = find.byType(FlatButton);
-    expect(tester.widget<FlatButton>(flatButton).enabled, false);
-  });
 
-  testWidgets('FlatButton onPressed and onLongPress callbacks are distinctly recognized', (WidgetTester tester) async {
-    bool didPressButton = false;
-    bool didLongPressButton = false;
+    expect(tester.getSize(find.byType(FlatButton)), equals(const Size(88.0, 48.0)));
```

Comment:
It looks like you might have deleted two newer preexisting tests on `flat_button_test.dart` by mistake: 
- 'FlatButton onPressed and onLongPress callbacks are correctly called when non-null'
- 'FlatButton onPressed and onLongPress callbacks are distinctly recognized'